react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(undefined);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    const readMe = async () => {
+      setIsLoading(true)
+      const data = await readItems();
+      setItems(data);
+      setIsLoading(false);
+    }
+    if (isLoading) {
+      readMe().catch((err) => {
+      setError(err);
+      setIsLoading(false)
+      })
+    }
+    if (isLoading === undefined) {
+      setIsLoading(() => !isLoading)
```

Comment:
Or just `true` since you know that's what the value will be. Takes less thought for the reader.